scalyr_agent/__scalyr__.py

```diff
@@ -155,13 +195,35 @@ def get_install_root():
     @rtype: six.text_type
     """
     # See the listed cases above.  From that, it should be clear that these rules work for the different cases.
-    parent_of_package_install = os.path.dirname(get_package_root())
-    if __is_frozen__:  # win32 install
-        return parent_of_package_install
-    elif os.path.basename(parent_of_package_install) != "py":  # Running from Source
-        return parent_of_package_install
-    else:  # Installed using tarball or rpm/debian package
-        return os.path.dirname(parent_of_package_install)
+
+    # On dev install just return source root as install root.
+    if INSTALL_TYPE == DEV_INSTALL:
+        package_root = os.path.dirname(__file__)
+        return os.path.dirname(package_root)
+
+    if __is_frozen__:
+        # In case of frozen binary, the sys.executable is the frozen binary itself
+        executable_path = sys.executable
+    else:
+        executable_path = __file__
+
+    # Resolve possible symlinks until we reach the real file.
+    while os.path.islink(executable_path):
+        executable_path = os.path.realpath(executable_path)
+
+    if __is_frozen__:
+        # In case if frozen binaries the executable has to be located in the '<install root>/bin', so we just
+        # go down until we reach the the install root.
```

Comment:
I guess here it should be "go up" instead of "go down", since if I understand it correctly, we go to the parent directory of bin/, right?

Reply:
Yeah, I guess that's more correct to have root at the top)